includes/Core/Modules/Module.php

```diff
@@ -337,7 +339,7 @@ public function is_connected() {
 			} catch ( Exception $e ) {
 				// Set every result of this batch to the exception.
 				foreach ( $results as $key => $datapoint ) {
-					if ( ! is_string( $datapoint ) || ! isset( $datapoint_services[ $datapoint ] ) || $service_identifier !== $datapoint_services[ $datapoint ] ) {
+					if ( ! is_string( $datapoint ) || ! isset( $datapoint_definitions[ $datapoint_name ] ) || $service_identifier !== $datapoint_definitions[ $datapoint_name ]['service'] ) {
```

Comment:
This suggestion removes the [duplicate `isset` check](https://user-images.githubusercontent.com/1621608/86273152-7d6f5600-bbd8-11ea-9ddf-53f021fd4701.png) (not your addition 😄 ) and refactors with `$datapoint_service` 

```suggestion
					if ( ! is_string( $datapoint ) || $service_identifier !== $datapoint_service ) {
```

Reply:
Reworked it in another way because `$datapoint_service` is defined in another loop and we can't rely on it.